src/VisualStudio/Core/Next/Remote/JsonRpcClient.cs

```diff
@@ -81,9 +81,8 @@ public Task<T> InvokeAsync<T>(string targetName, IEnumerable<object> arguments,
 
         public void Dispose()
         {
-            OnDisposed();
-
-            _rpc.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
I don't think this has finalizer. this pattern is right for the one that has finalizer but I am using IDisposble for C# RAII pattern, not for the other disposable pattern.